src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -49,17 +51,41 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
    */
   private final static def manifestLocation = "/checkerframework/"
 
+  /**
+   * Configure each task in {@code project} with the given {@code taskType}.
+   * <p>
+   * We prefer to configure with {@link
+   * org.gradle.api.tasks.TaskCollection#configureEach(org.gradle.api.Action)}
+   * rather than {@link
+   * org.gradle.api.tasks.TaskCollection#all(org.gradle.api.Action)}, but {@code
+   * configureEach} is only available on Gradle 4.9 and newer, so this method
+   * dynamically picks the better candidate based on the current Gradle version.
+   * <p>
+   * See also: <a href="https://docs.gradle.org/current/userguide/task_configuration_avoidance.html">
+   * Gradle documentation: Task Configuration Avoidance</a>
+   */
+  private static <S extends Task> void configureTasks(Project project, Class<S> taskType, Action<? super S> configure) {
+    def (major, minor) = project.gradle.gradleVersion.split('\\.').collect { Integer.parseInt(it) }
+    if (major < 4 || (major == 4 && minor < 9)) {
```

Comment:
How about the following? (I'm not sure `<` works in Groovy, if it does not, then `compareTo` should work)
```suggestion
    if (GradleVersion.current() < GradleVersion.version("4.9")) {
```